npm_and_yarn/spec/dependabot/npm_and_yarn/update_checker_spec.rb

```diff
@@ -675,6 +675,62 @@
     end
   end
 
+  describe "#lowest_resolvable_security_fix_version" do
+    subject(:lowest_resolvable_security_fix_version) { checker.lowest_resolvable_security_fix_version }
+
+    let(:dependency_files) { project_dependency_files("npm8/locked_transitive_dependency") }
+    let(:dependency_name) { "@dependabot-fixtures/npm-transitive-dependency" }
+    let(:target_version) { "1.2.1" }
+    let(:dependency) do
+      Dependabot::Dependency.new(
+        name: dependency_name,
+        version: "1.0.0",
+        requirements: [],
+        package_manager: "npm_and_yarn"
+      )
+    end
+
+    context "when the dependency is not vulnerable" do
+      before { allow(checker).to receive(:vulnerable?).and_return(false) }
```

Comment:
It's a bad practice to mock the class you're testing.

I would suggest looking at another test in this file that uses `@dependabot-fixtures`. Create a minimal reproduction of the `NoChangeError` in that org and use it in this test to verify `lowest_resolvable_security_fix_version` returns `nil`.

Reply:
fixed